### PR TITLE
Add Rosenthal temperature source model

### LINKED ISSUE #63: Add Rosenthal meltpool temperature profile
## Reason
Analytical model for providing meltpool profile and temperature for a given laser power and scanning speed.

## Design
Material that implements the thermal profile and outputs meltpool depth/width. 

## Impact
A new simplified capability for meltppol geometry, enables connection to microstructural and other material models.

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
Closes #63

### CODE REVIEW

**Comment by SudiptaBiswas** [comment]
Please review @lindsayad @jiangwen84

File: src/materials/RosenthalTemperatureSource.C [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/*                                                                          */
+/* MALAMUTE: MOOSE Application Library for Advanced Manufacturing UTilitiEs */
+/*                                                                          */
+/*           Copyright 2021 - 2023, Battelle Energy Alliance, LLC           */
+/*                           ALL RIGHTS RESERVED                            */
+/****************************************************************************/
+
+#include "RosenthalTemperatureSource.h"
+
+// #include "Function.h"
```
Comment (jiangwen84) [inline src/materials/RosenthalTemperatureSource.C]: delete it?

File: test/tests/melt_pool_geometry/rosenthal_temp_source.i [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+[Preconditioning]
+  [full]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = PJFNK
```
Comment (jiangwen84) [inline test/tests/melt_pool_geometry/rosenthal_temp_source.i]: NEWTON?

File: test/tests/melt_pool_geometry/rosenthal_temp_source.i [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+[Variables]
+  [temp]
+    initial_condition = 393
+  []
+[]
+
+[AuxVariables]
+  [temp_src]
+    order = FIRST
+    family = MONOMIAL
```
Comment (jiangwen84) [inline test/tests/melt_pool_geometry/rosenthal_temp_source.i:20]: why to use first monomial?

**Review by jiangwen84 — CHANGES_REQUESTED** [review]
Looks good.

**Comment by jiangwen84** [comment]
@SudiptaBiswas  do you know why those tests are pending?

**Comment by lindsayad** [comment]
Ok they should activate automatically in the future

File: test/tests/melt_pool_geometry/rosenthal_temp_source.i (hunk 3 above)
Comment (SudiptaBiswas) [reply test/tests/melt_pool_geometry/rosenthal_temp_source.i:20]: For interpolating the material property and allowing contour plots.

File: test/tests/melt_pool_geometry/tests [hunk 4]
```diff
@@ -0,0 +1,19 @@
+[Tests]
+  issues = '#63'
+  [rosenthal_temp]
+    type = 'Exodiff'
+    input = 'rosenthal_temp_source.i'
+    exodiff = 'rosenthal_temp_source_out.e'
+    abs_zero = 1e-8
+    design = 'RosenthalTemperatureSource.md'
+    requirement = 'MALAMUTE shall provide analytical expression for meltpool geometry and corresponding temperature profile'
```
Comment (cticenhour) [inline test/tests/melt_pool_geometry/tests]:
```suggestion
    requirement = 'The system shall provide analytical expressions for meltpool geometry and corresponding temperature profile'
```

This wording better complies with MOOSE application SQA standards. Also, corrected what I thought was a typo.

File: test/tests/melt_pool_geometry/tests [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  []
+  [rosenthal_depth]
+    type = 'CSVDiff'
+    input = 'rosenthal_temp_source.i'
+    csvdiff = 'rosenthal_temp_source_out.csv'
+    abs_zero = 1e-8
+    design = 'RosenthalTemperatureSource.md'
+    requirement = 'MALAMUTE shall provide analytical expression for meltpool depth and width'
```
Comment (cticenhour) [inline test/tests/melt_pool_geometry/tests]:
```suggestion
    requirement = 'The system shall provide analytical expressions for meltpool depth and width'
```

Same as previous comment.

**Review by cticenhour — CHANGES_REQUESTED** [review]

**Review by jiangwen84 — APPROVED** [review]

**Review by cticenhour — APPROVED** [review]

**Comment by cticenhour** [comment]
@jiangwen84 you should already have merge rights on MALAMUTE. Please use this PR to test your permissions.

**Comment by jiangwen84** [comment]
@cticenhour yes, it works for me. thanks. 

@SudiptaBiswas  let me know if you would like to try the nodal patch recover on the temperature field. If not, I can merge it for you.

**Comment by jiangwen84** [comment]
@SudiptaBiswas  FYI https://mooseframework.inl.gov/source/auxkernels/NodalPatchRecoveryAux.html

**Comment by SudiptaBiswas** [comment]
> @SudiptaBiswas let me know if you would like to try the nodal patch recover on the temperature field. If not, I can merge it for you.

This can be merged as is. I will add additional tests later.

**Pull request merged**